lib/models/task.dart

```diff
@@ -138,8 +138,11 @@ class Task {
     DateTime? created,
     DateTime? updated,
     DateTime? dueDate,
+    bool? clearDueDate,
     DateTime? startDate,
+    bool? clearStartDate,
     DateTime? endDate,
+    bool? clearEndDate,
```

Comment:
This allows for clearing the dates by setting them to null if the corresponding bool is true.